src/widgets/scale/lv_scale.h

```diff
@@ -169,6 +171,25 @@ void lv_scale_set_range(lv_obj_t * obj, int32_t min, int32_t max);
  */
 void lv_scale_set_round_props(lv_obj_t * obj, uint32_t angle_range, int32_t rotation);
 
+/**
+ * Point the needle to the corresponding value through the line
+ * @param obj              pointer to a scale object
+ * @param needle_line      needle_line of the scale
+ * @param needle_length    length of the needle
```

Comment:
> Just a small idea: It would be useful if negative value meant "make the line smaller than the radius of the scale by this value". This way we could easily create responsive scales. E.g.
> 
> ```
> lv_obj_set_size(scale, 100, 100); //radius is 50
> lv_scale_set_line_needle_value(scale, line, -20, value); //needle_length = 50-20 = 30
> ```
> 
> What do you think?

When value is greater than 0, needle_length equals value; When value is less than 0, needle_length equals "radius -value". Do you mean that?

Reply:
Exactly :slightly_smiling_face:

Reply:
> Exactly

OK